exchanges/okx/okx_websocket.go

```diff
@@ -0,0 +1,2155 @@
+
+// InstrumentsSubscription to subscribe for instruments. The full instrument list will be pushed
+// for the first time after subscription. Subsequently, the instruments will be pushed if there is any change to the instrument’s state (such as delivery of FUTURES,
+// exercise of OPTION, listing of new contracts / trading pairs, trading suspension, etc.).
+func (ok *Okx) InstrumentsSubscription(operation string, assetType asset.Item, pair currency.Pair) (*SubscriptionOperationResponse, error) {
+	return ok.WsChannelSubscription(operation, okxChannelInstruments, assetType, pair, true)
+}
+
+// TickersSubscription subscribing to "ticker" channel to retrieve the last traded price, bid price, ask price and 24-hour trading volume of instruments. Data will be pushed every 100 ms.
+func (ok *Okx) TickersSubscription(operation string, assetType asset.Item, pair currency.Pair) (*SubscriptionOperationResponse, error) {
+	return ok.WsChannelSubscription(operation, okxChannelTickers, assetType, pair, false, true)
+}
+
+// OpenInterestSubscription to subscribe or unsubscribe to "open-interest" channel to retrieve the open interest. Data will by pushed every 3 seconds.
+func (ok *Okx) OpenInterestSubscription(operation string, assetType asset.Item, pair currency.Pair) (*SubscriptionOperationResponse, error) {
+	if !(assetType == asset.Futures || assetType == asset.Option || assetType == asset.PerpetualSwap) {
+		return nil, fmt.Errorf("%w, only FUTURES, SWAP and OPTION asset types are supported", errInvalidInstrumentType)
+	}
+	return ok.WsChannelSubscription(operation, okxChannelOpenInterest, assetType, pair, false, true)
+}
+
+// CandlesticksSubscription to subscribe or unsubscribe to "candle" channels to retrieve the candlesticks data of an instrument. the push frequency is the fastest interval 500ms push the data.
+func (ok *Okx) CandlesticksSubscription(operation, channel string, assetType asset.Item, pair currency.Pair) (*SubscriptionOperationResponse, error) {
+	if !(channel == okxChannelCandle1Y || channel == okxChannelCandle6M || channel == okxChannelCandle3M || channel == okxChannelCandle1M || channel == okxChannelCandle1W || channel == okxChannelCandle1D || channel == okxChannelCandle2D || channel == okxChannelCandle3D || channel == okxChannelCandle5D || channel == okxChannelCandle12H || channel == okxChannelCandle6H || channel == okxChannelCandle4H || channel == okxChannelCandle2H || channel == okxChannelCandle1H || channel == okxChannelCandle30m || channel == okxChannelCandle15m || channel == okxChannelCandle5m || channel == okxChannelCandle3m || channel == okxChannelCandle1m || channel == okxChannelCandle1Yutc || channel == okxChannelCandle3Mutc || channel == okxChannelCandle1Mutc || channel == okxChannelCandle1Wutc || channel == okxChannelCandle1Dutc || channel == okxChannelCandle2Dutc || channel == okxChannelCandle3Dutc || channel == okxChannelCandle5Dutc || channel == okxChannelCandle12Hutc || channel == okxChannelCandle6Hutc) {
```

Comment:
```suggestion
	if channel != okxChannelCandle1Y && channel != okxChannelCandle6M && channel != okxChannelCandle3M && channel != okxChannelCandle1M && channel != okxChannelCandle1W && channel != okxChannelCandle1D && channel != okxChannelCandle2D && channel != okxChannelCandle3D && channel != okxChannelCandle5D && channel != okxChannelCandle12H && channel != okxChannelCandle6H && channel != okxChannelCandle4H && channel != okxChannelCandle2H && channel != okxChannelCandle1H && channel != okxChannelCandle30m && channel != okxChannelCandle15m && channel != okxChannelCandle5m && channel != okxChannelCandle3m && channel != okxChannelCandle1m && channel != okxChannelCandle1Yutc && channel != okxChannelCandle3Mutc && channel != okxChannelCandle1Mutc && channel != okxChannelCandle1Wutc && channel != okxChannelCandle1Dutc && channel != okxChannelCandle2Dutc && channel != okxChannelCandle3Dutc && channel != okxChannelCandle5Dutc && channel != okxChannelCandle12Hutc && channel != okxChannelCandle6Hutc {
```

This would be much more readable with a map ok check